static/app/components/forms/compactSelect.tsx

```diff
@@ -125,6 +130,82 @@ export const CompactSelectControl = ({
   );
 };
 
+// TODO(vl): Turn this into a reusable component
+function Menu({
+  // Trigger & trigger state
+  targetRef,
+  onClose,
+  // Overlay props
+  offset = 8,
+  crossOffset = 0,
+  containerPadding = 8,
+  placement = 'bottom left',
+  shouldCloseOnBlur = true,
+  isDismissable = true,
+  maxMenuHeight = 400,
+  children,
+}: MenuProps) {
```

Comment:
Moving the menu out into a separate component.

**The problem**: when the viewport isn't tall enough and we open the menu, then close it, then change the viewport height (e.g. by resizing the browser window), then open the menu again, it will render at the previous height from before the viewport resize. This happens because `useOverlayPosition` still remembers the viewport size from before and doesn't know that the viewport has changed.
<img width="343" alt="image" src="https://user-images.githubusercontent.com/44172267/175750307-ebd2c47e-e190-46c5-bd92-97d1897f1884.png">

**The solution**: move the menu, including `useOverlayPosition` and related hooks, out into its own component. This menu component will get rendered/destroyed every time `state.isOpen` changes. As such, `useOverlayPosition` will get reinitialized every time we open the menu, ensuring that it always has a fresh reference to the viewport size.